client.c

```diff
@@ -1864,12 +1864,12 @@ static void request_client_etas(struct client_ops *ops)
 static int handle_cmd_timeout(struct fio_client *client,
 			      struct fio_net_cmd_reply *reply)
 {
-	flist_del(&reply->list);
-	free(reply);
-
 	if (reply->opcode != FIO_NET_CMD_SEND_ETA)
 		return 1;
 
+	flist_del(&reply->list);
```

Comment:
I'm not sure this is correct. Before we always killed a reply, which means pruning from the list and freeing it. Now we leave the non-ETA commands on there. How about something ala:

```
@@ -1849,10 +1849,12 @@ static void request_client_etas(struct client_ops *ops)
 static int handle_cmd_timeout(struct fio_client *client,
                              struct fio_net_cmd_reply *reply)
 {
+       uint16_t op = reply->opcode;
+
        flist_del(&reply->list);
        free(reply);
 
-       if (reply->opcode != FIO_NET_CMD_SEND_ETA)
+       if (op != FIO_NET_CMD_SEND_ETA)
                return 1;

```

Reply:
Updated this way.